packages/button/src/Button.ts

```diff
@@ -127,11 +144,33 @@ export class Button extends SizedMixin(ButtonBase, { noDefaultSize: true }) {
     public set quiet(quiet: boolean) {
         this.treatment = quiet ? 'outline' : 'fill';
     }
-
     public get quiet(): boolean {
         return this.treatment === 'outline';
     }
 
+    protected override willUpdate(
+        changedProperties: PropertyValues<this>
+    ): void {
+        super.willUpdate(changedProperties);
+        if (changedProperties.has('pending')) {
+            if (this.pending) {
+                if (!this.disabled) {
+                    this.setAttribute('aria-label', this.pendingLabel);
+                }
+            } else if (
+                changedProperties.get('pending') !== undefined &&
+                !this.pending
+            ) {
+                this.setAttribute('aria-label', this.label || '');
+            }
+        }
+        if (changedProperties.has('disabled')) {
+            if (this.pending && !this.disabled) {
+                this.setAttribute('aria-label', this.pendingLabel);
+            }
```

Comment:
`if (this.pending && this.disabled)` should we remove the `aria-label` in this case?

Reply:
right yes!